### PR TITLE
Removes 200mb of wasted memory usage

### PR DESCRIPTION
**Pull request by MrsTonedOne**
>Hey everybody! Lets store a copy of a hashlist containing all 357 mob/living subtypes on all 15 thousand machines on the map. (remember, pipes are machinery)
-coiax

@tgstation/commit-access @coiax @vuonojenmustaturska

### CODE REVIEW

**Comment by RandomMarine** [comment]
Wouldn't recovering 200mb of memory be the correct term?

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -485,6 +485,7 @@
 
 //Picks from the list, with some safeties, and returns the "default" arg if it fails
 #define DEFAULTPICK(L, default) ((islist(L) && length(L)) ? pick(L) : default)
+#define THISORTHAT(THIS, THAT) (THIS ? THIS : THAT)
```
Comment (duncathan) [inline code/__HELPERS/_lists.dm]: why on earth do we need a define for an operator

Comment (duncathan) [reply code/__HELPERS/_lists.dm]: and what is it doing in the list defines

Comment (duncathan) [reply code/__HELPERS/_lists.dm]: ALSO just fucking use `this || that` lmao
but not as a define that's dumb

Comment (MrStonedOne) [reply code/__HELPERS/_lists.dm]: to understand this you have to understand the thought proces.s.

I wanted the equivalent of defaultpick, but for just accessing lists, but then i went to find the usage to add it in, and i noticed that it didn't actually access an index it passed it to istypeintypecache, so i made it for reading a variable, not accessing an index.

Comment (optimumtact) [reply code/__HELPERS/_lists.dm]: it's clearer

**Comment by AnturK** [comment]
Good stuff, but that dunc said that define is overkill

File: code/__HELPERS/_lists.dm (hunk 1 above)
Comment (MrStonedOne) [reply code/__HELPERS/_lists.dm]: anywho, I plan on keeping it, it's easier to read and follow chains of this compared to ? that I actually want to ban the use of ? now and make everything use thisorthat

Comment (duncathan) [reply code/__HELPERS/_lists.dm]: use `this || that` since it accomplishes what you want and kill this

**Review by duncathan — CHANGES_REQUESTED** [review]
see above

**Review by duncathan — APPROVED** [review]

File: code/__HELPERS/_lists.dm [hunk 2]
```diff
@@ -485,6 +485,7 @@
 
 //Picks from the list, with some safeties, and returns the "default" arg if it fails
 #define DEFAULTPICK(L, default) ((islist(L) && length(L)) ? pick(L) : default)
+
```
Comment (ShizCalev) [inline code/__HELPERS/_lists.dm]: blank

**Review by optimumtact — APPROVED** [review]

File: code/game/machinery/_machinery.dm [hunk 3]
```diff
@@ -189,7 +190,7 @@ Class Procs:
 	density = TRUE
 	if(!target)
 		for(var/am in loc)
-			if(!is_type_in_typecache(am, occupant_typecache))
+			if(!is_type_in_typecache(am, (occupant_typecache || GLOB.typecache_living)))
```
Comment (kevinz000) [inline code/game/machinery/_machinery.dm]: wouldn't this allow all living things in thigns that aren't meant to have all living things in it?

**Comment by JamieH** [comment]
FOUND BY HIPPIESTATION WEW

**Comment by banthebantz** [comment]
god damn it

**Comment by Cyberboss** [comment]
Canfricturrrrr

File: code/game/machinery/_machinery.dm (hunk 3 above)
Comment (nicbn) [reply code/game/machinery/_machinery.dm]: No because || operator is short circuited

**Pull request merged**